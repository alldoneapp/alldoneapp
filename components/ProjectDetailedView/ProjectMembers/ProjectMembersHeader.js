import React  from 'react'
import { StyleSheet, Text, View } from 'react-native'
import PropTypes from 'prop-types'
import styles, { colors } from '../../styles/global'
import { translate } from '../../../i18n/TranslationService'

const ProjectMembersHeader = ({ amount }) => {
    const parseText = number => {
        if (number == null || number <= 0) {
            return translate('No members yet')
        } else if (number > 1) {
            return translate('Number members', { number })
        }
        return translate('Number member', { number })
    }

    return (
        <View style={localStyles.container}>
            <Text style={[styles.title6, { color: colors.Text01 }]}>{translate('Project members')}</Text>
            <View style={localStyles.headerCaption}>
                <Text style={[styles.caption2, { color: colors.Text02 }]}>{parseText(amount)}</Text>
            </View>
        </View>
    )
}

ProjectMembersHeader.propTypes = {
    amount: PropTypes.number.isRequired,
}

const localStyles = StyleSheet.create({
    container: {
        height: 72,
        paddingTop: 32,
        paddingBottom: 12,
        alignItems: 'flex-end',
        flexDirection: 'row',
    },
    headerCaption: {
        marginLeft: 16,
        height: 22,
        justifyContent: 'center',
    },
})

export default ProjectMembersHeader
