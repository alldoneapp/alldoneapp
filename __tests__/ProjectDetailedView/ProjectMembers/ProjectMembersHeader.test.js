/**
 * @jest-environment jsdom
 */

import React from 'react'
import ProjectMembersHeader from '../../../components/ProjectDetailedView/ProjectMembers/ProjectMembersHeader'
import renderer from 'react-test-renderer'
import store from '../../../redux/store'
import { storeLoggedUserProjects } from '../../../redux/actions'

describe('ProjectMembersHeader component', () => {
    describe('ProjectMembersHeader snapshot test', () => {
        it('should render correctly', () => {
            store.dispatch(storeLoggedUserProjects([{ name: 'My Project', userIds: [] }]))
            const tree = renderer.create(<ProjectMembersHeader amount={0} />).toJSON()
            expect(tree).toMatchSnapshot()
        })
    })
})
