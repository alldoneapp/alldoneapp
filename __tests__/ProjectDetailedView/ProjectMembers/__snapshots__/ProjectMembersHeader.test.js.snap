// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectMembersHeader component ProjectMembersHeader snapshot test should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-end",
      "flexDirection": "row",
      "height": 72,
      "paddingBottom": 12,
      "paddingTop": 32,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#04142F",
          "fontFamily": "Roboto-Medium",
          "fontSize": 20,
          "lineHeight": 28,
        },
        Object {
          "color": "#04142F",
        },
      ]
    }
  >
    Team members
  </Text>
  <View
    style={
      Object {
        "height": 22,
        "justifyContent": "center",
        "marginLeft": 16,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#04142F",
            "fontFamily": "Roboto-Regular",
            "fontSize": 12,
            "letterSpacing": 0.48,
            "lineHeight": 20,
          },
          Object {
            "color": "#4E5D78",
          },
        ]
      }
    >
      No members yet
    </Text>
  </View>
</View>
`;
